cnxarchive/views.py

```diff
@@ -287,11 +287,17 @@ def _get_content_json(request=None, ident_hash=None, reqtype=None):
     except IdentHashSyntaxError:
         raise httpexceptions.HTTPNotFound()
 
+    page_ident_hash = routing_args.get('page_ident_hash', '')
+    if page_ident_hash:
+        try:
+            p_id, p_version, p_id_type = split_ident_hash(page_ident_hash,
+                                                          return_type=True)
+        except IdentHashSyntaxError:
+            raise httpexceptions.HTTPNotFound()
```

Comment:
This short circuits a bad page id (integer page numbers are handled in webview itself) and sets up the availability of the page ident parts (p_*)